DOCS-4447 Adds Create an API Integration Documentation (#13354)
DataDog/integrations-core: docs/dev/api_integration.md

   {{< code-block lang="shell" >}}cd $HOME/dd/integrations-extras{{< /code-block >}}

2. Set up a Python virtual environment:
   {{< code-block lang="shell" >}}
   python3 -m venv venv
   . venv/bin/activate{{< /code-block >}}

   You can exit the virtual environment at any time by running `deactivate`.

3. Install the [Developer Toolkit][19]:
   {{< code-block lang="shell" >}}pip3 install "datadog-checks-dev[cli]"{{< /code-block >}}

   If you are using the Z Shell, you may need to use escaped characters by running `pip3 install datadog-checks-dev\[cli\]`.

4. Set `integrations-extras` as the default working repository:
   {{< code-block lang="shell" >}}
   ddev config set integrations-extras $HOME/dd/integrations-extras
   ddev config set repo integrations-extras{{< /code-block >}}

   If you used a directory other than `$HOME/dd` to clone the `integrations-extras` directory, use the following command to set your working repository:

   {{< code-block lang="shell" >}}
   ddev config set integrations-extras <PATH/TO/INTEGRATIONS-EXTRAS>
   ddev config set repo integrations-extras{{< /code-block >}}

## Populate the integration tile scaffolding

> Run the `ddev` command to generate a skeleton of the folders and files needed for your integration. The options you use with the command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see [Integrations assets][20].
> 
> ### Create an informational tile only listing
> 
> To create the informational tile-only listing's scaffolding:
> 
> 1. Make sure you're inside the `integrations-extras` directory:
>    {{< code-block lang="shell" >}}cd $HOME/dd/integrations-extras{{< /code-block >}}
> 2. Run the `ddev` command with the `-t tile` option
>    {{< code-block lang="shell" >}}ddev create -t tile "<Offering Name>"{{< /code-block >}}

Reviewer: ```suggestion
Run the `ddev` command to create scaffolding for an informational tile-only listing:

1. Make sure you're inside the `integrations-extras` directory:
   {{< code-block lang="shell" >}}cd $HOME/dd/integrations-extras{{< /code-block >}}
2. Run the `ddev` command with the `-t tile` option
   {{< code-block lang="shell" >}}ddev create -t tile "<Offering Name>"{{< /code-block >}}
   
The options you use with the `ddev` command are different depending on what type of integration you are developing. For a full list of the files created by the `ddev` command, see [Integrations assets][20].   
```
I think it's clearer just to use the one section here.

---
title: Create an API Integration
type: documentation
---

## Overview

> You can use [Datadog API endpoints][1] to programmatically send data to Datadog, manage your account, and visualize data collected from an API integration. 

Reviewer 2: ```suggestion
Use an [Datadog API endpoints][1] to enrich and submit data from your backend, or pull data directly out of Datadog. API integrations work well in building a connector between Datadog and another SaaS platform. This method is ideal for Technology Partners that are SaaS based, and have an existing website for users to log into for authorization purposes.
```

API integrations send the following types of data to Datadog:

- [Metrics][12]
- [Logs & Log Pipelines][13]
- [Events][14]
- [Service Checks][15]
- [Traces][16]
- [Incidents][17]
- [Security Events][18]

> This page provides instructions for creating an API integration in the `integrations-extras` repository. For more information about why you would want to create an API-based integration, see [Creating your own solution][2].

Reviewer 2: ```suggestion
This page provides instructions for creating an API integration in the `integrations-extras` repository. For more information about why you would want to create an API-based integration, see [Creating your own solution][2]. Since API integrations do not use the Datadog Agent to collect data, you need to create an informational tile-only listing once your development work is complete. 
```

## Setup

> ### Prerequisites
> 
> - You must have an [API key][3] and [application key][4].
> - Determine which [Datadog site][5] you want to use.
> 
> An API key is required to submit data to a Datadog API endpoint. An application key is required to query data from Datadog or to create resources within the Datadog site. For more information, see [API and Application Key][6].
> 
> Create a connection to Datadog in your company's platform using the API key, application key, and site URL. 
> 
> ### Create an OAuth client
> 
> Datadog recommends setting up an [OAuth client][7] to securely authorize third-party application access in your integrations. For more information, see [OAuth for Integrations][8] and [Authorization Endpoints][9].

Reviewer 2: ```suggestion
### Prerequisites

- You must have an [API key][3] and [application key][4].
- Determine which [Datadog site][5] you want to use.

An API key is required to submit data to a Datadog API endpoint. An application key is required to query data from Datadog or to create resources within the Datadog site. For more information, see [API and Application Key][6].

Create a connection to Datadog in your company's platform using the API key, application key, and site URL. 

### Create an OAuth client
Instead of requesting these credentials directly from a user, Datadog recommends using an [OAuth client][7] to handle authorization and access for API-based integrations. For more information, see [OAuth for Integrations][8] and [Authorization Endpoints][9]

You can explore examples of existing API integrations in the `integrations-extras` repository such as [Vantage][23].
```

Follow this template to define the `media` object in the media carousel which includes an image, a video thumbnail, and a video:

{{< code-block lang="json" filename="manifest.json" collapsible="true" >}}
"media": [
      {
        "media_type": "image",
        "caption": "A Datadog API Integration OOTB Dashboard",
        "image_url": "images/integration_name_image_name.png"
      },
      {
        "media_type": "video",
        "caption": "A Datadog API Integration Overview Video",
        "image_url": "images/integration_name_video_thumbnail.png",
        "vimeo_id": 123456789
      },
    ],
{{< /code-block >}}

For more information, see [Integrations Assets Reference][21].

## Open a pull request

Open a pull request in the [`integrations-extras` repository][10] that adds images (such as logos and images) and asset files (such as `Changelog.md`, `README.md`, and `manifest.json`) to your API integration's tile-only listing in the [Integrations page][11]. Automatic tests run checks to verify that your pull request is in good shape and contains all the required content to be updated.

## Review process

> Once your pull request passes all the checks, reviewers from the `Datadog/agent-integrations` and `Datadog/documentation` teams provide suggestions and feedback on best practices.

Reviewer 2: ```suggestion
Once your pull request passes all the checks, reviewers from the `Datadog/agent-integrations`, `Datadog/marketplace` and `Datadog/documentation` teams provide suggestions and feedback on best practices.
```
Not sure if that label actually gets added but someone from our team reviews the PR as well.

[3]: https://docs.datadoghq.com/account_management/api-app-keys/#api-keys
[4]: https://docs.datadoghq.com/account_management/api-app-keys/#application-keys
[5]: https://docs.datadoghq.com/getting_started/site
[6]: https://docs.datadoghq.com/account_management/api-app-keys/
[7]: https://docs.datadoghq.com/developers/authorization/
[8]: https://docs.datadoghq.com/developers/integrations/oauth_for_integrations/
[9]: https://docs.datadoghq.com/developers/authorization/oauth2_endpoints/
[10]: https://github.com/DataDog/integrations-extras/
[11]: https://app.datadoghq.com/integrations
[12]: https://docs.datadoghq.com/api/latest/metrics/
[13]: https://docs.datadoghq.com/logs/faq/partner_log_integration/
[14]: https://docs.datadoghq.com/api/latest/events/
[15]: https://docs.datadoghq.com/api/latest/service-checks/
[16]: https://docs.datadoghq.com/tracing/guide/send_traces_to_agent_by_api/
[17]: https://docs.datadoghq.com/api/latest/incidents/
[18]: https://docs.datadoghq.com/api/latest/security-monitoring/
[19]: https://www.python.org/downloads/
[20]: https://pypi.org/project/datadog-checks-dev/
[21]: https://docs.datadoghq.com/developers/integrations/check_references/#manifest-file

> [22]: https://datadoghq.com/blog/

Reviewer 2: ```suggestion
[22]: https://datadoghq.com/blog/
[23]: https://github.com/DataDog/integrations-extras/tree/master/vantage
```

        "caption": "A Datadog API Integration OOTB Dashboard",
        "image_url": "images/integration_name_image_name.png"
      },
      {
        "media_type": "video",
        "caption": "A Datadog API Integration Overview Video",
        "image_url": "images/integration_name_video_thumbnail.png",
        "vimeo_id": 123456789
      },
    ],
{{< /code-block >}}

For more information, see [Integrations Assets Reference][21].

## Open a pull request

Open a pull request in the [`integrations-extras` repository][10] that adds images (such as logos and images) and asset files (such as `Changelog.md`, `README.md`, and `manifest.json`) to your API integration's tile-only listing in the [Integrations page][11]. Automatic tests run checks to verify that your pull request is in good shape and contains all the required content to be updated.

## Review process

Once your pull request passes all the checks, reviewers from the `Datadog/agent-integrations`, `Datadog/marketplace`, and `Datadog/documentation` teams provide suggestions and feedback on best practices.

Once you have addressed the feedback and re-requested reviews, these reviewers approve your pull request.

> ## Coordinate GTM opportunities
> 
> Once an Integrations tile is live, Technology Partners can meet with Datadog's Partner Marketing team to coordinate a joint go-to-market (GTM) strategy, which includes the following:
> 
> - A Datadog quote for partner press releases
> - A blog post on the [Datadog Monitor][22]
> - Amplification of social media posts

Reviewer 2: This can be removed since it doesn't apply to free integrations (aka ones in integrations-extras)

```suggestion
```
